pkg/client/api.go

```diff
@@ -81,21 +81,39 @@ func (c *httpAPIClient) Do(ctx context.Context, verb, endpoint string, query url
 	}()
 
 	if err != nil {
-		return APIResponse{}, err
+		return APIResponse{}, &Error{
+      Type: ErrBadResponse,
+      Msg: fmt.Sprintf("HTTP request failed [Method: %s] [URL: %s] [Headers: %+v] [Body: %s] [Error: %v]",
```

Comment:
Only include actionable info here, e.g. headers is not useful to us because we already know from prom adapter logs that the source is prom adapter